packages/pages-components/src/map/coordinate.ts

```diff
@@ -130,16 +132,21 @@ class Coordinate {
    * and one one {@link LONGITUDE_ALIASES | longitude alias}.
    * @param longitude - Optional only if the first argument is a {@link Coordinate}-like object
    */
-  constructor(latitudeOrObject: number | object, longitude?: number) {
-    const latitude = latitudeOrObject;
+  constructor(latitudeOrObject: number | object, long?: number) {
+    let latitude = latitudeOrObject;
+    let longitude: number | latLngFunction | undefined = long;
 
     this._lat = 0;
     this._lon = 0;
 
-    if (typeof latitudeOrObject === "object") {
-      this._lat = findValue(latitudeOrObject, LATITUDE_ALIASES) ?? 0;
-      this._lon = findValue(latitudeOrObject, LONGITUDE_ALIASES) ?? 0;
-    } else if (typeof latitude === "number" && longitude) {
+    if (typeof latitudeOrObject == "object") {
+      latitude = findValue(latitudeOrObject, LATITUDE_ALIASES) ?? 0;
+      longitude = findValue(latitudeOrObject, LONGITUDE_ALIASES) ?? 0;
+
+      this.latitude = typeof latitude == "function" ? latitude() : latitude;
```

Comment:
Don't you want to set this._lat and this._lon? I'm actually a little confused because some of the other functions in this class use this._lat and some use this.longitude. Why are there two values? Should the _ versions just be removed and the two functions that use them instead operate off of this.latitude/this.longitude?

Reply:
https://www.typescripttutorial.net/typescript-tutorial/typescript-getters-setters/

Reply:
I think the setup is wrong though. There are two variables for each set, _lat (which is private) and latitude (which is public). Every function is using latitude EXCEPT for the getter/setter which use _lat. So I'm pretty sure unexpected things would happen. If a user calls setLatitude() only _lat is set, but then never used by other functions, so when they call those other functions they'll get back that data housed by the latitude variable.

Additionally, latitude is public, yet there's a getter that returns the private variable. Try out something like this:
```
let foo = new Coordinate(1,2);
foo.setLatitude(5);
console.log(foo.latitude); // prints 1
console.log(foo.getLatitude()); // prints 5
```

The code was already weird like this but I'm pretty sure it's a bug.

Reply:
I think I get what you're saying but not sure if I understand how to fix it - should everything use this.latitude and completely get rid of this._lat?

Reply:
Yes, use this.latitude everywhere instead.